### PR TITLE
Add support for --config-path style cli option

### PR DESCRIPTION
**Pull request by Stonesjtu**
## Motivation

Support standard --xxx-xx style python cli option.

The original --xxx_xx is also supported automatically by argparser.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebookresearch/hydra/blob/master/CONTRIBUTING.md)?

Yes

## Test Plan

It's totally BC.
No special tests required.

## Related Issues and PRs

None

### CODE REVIEW

**Comment by omry** [comment]
Please fix the tests.
You can learn how how to run them locally here:

https://hydra.cc/docs/next/development/contributing

**Comment by omry** [comment]
Looks like it's not exactly backward compatible:

```
usage: my_app.py [--help] [--hydra-help] [--version] [--cfg {job,hydra,all}]
                 [--run] [--multirun] [--shell-completion]
                 [--config-path CONFIG_PATH] [--config-name CONFIG_NAME]
                 [overrides [overrides ...]]
my_app.py: error: unrecognized arguments: --config_name=cfg1 --config_path=dir1
________________ test_config_name_and_path_overrides[cfg1-dir2] ________________
```

**Comment by Stonesjtu** [comment]
Unfortunately, argparse does less than what I expected.

Can we make the BC breaking change or I'll just close the PR.

**Comment by omry** [comment]
Let's do it, but add a news fragment about this change (as api_change extension).

**Comment by omry** [comment]
Also look for references to the command line flags you are changing in the website docs directory and fix those too.

**Comment by Stonesjtu** [comment]
@omry ready for review. BTW I didn't find CLI options `bash_completion`,  `config_path` or `config_name` in the website docs.

File: news/661.api_change [hunk 1]
```diff
@@ -0,0 +1 @@
+Unify the CLI parser keyword to dash-style
```
Comment (omry) [inline news/661.api_change:1]: This sounds much bigger than what it actually is.
How about:
```suggestion
Hydra argparse flags are now consistently using dash-style.
```

**Comment by omry** [comment]
I touched related files, can you rebase?

File: news/661.api_change (hunk 1 above)
Comment (Stonesjtu) [reply news/661.api_change]: done

**Comment by Stonesjtu** [comment]
Rebased.

**Pull request merged**